pix2pix (#365)
tensorflow/swift-models: pix2pix/README.md

# pix2pix

> This repositery tries to closely resemble original pix2pix paper with the same architecture and hyper-parameters, but using Swift for TensorFlow. Results seem to be close to what original authors claimed to achieve, this is what a generator outputs after 7 hours of training using single GTX 1080 card.

Reviewer: ```suggestion
This example tries to closely resemble [the original pix2pix paper](https://arxiv.org/abs/1611.07004) with the same architecture and hyper-parameters, but using Swift for TensorFlow. Results seem to be close to what the original authors claimed to achieve, this is what a generator outputs after 7 hours of training using single GTX 1080 card:
```

![](https://i.imgur.com/Od9dfe8.jpg)
![](https://i.imgur.com/zQxPCAd.jpg)

In order to run the project you have to just call 

```bash

> swift run -c release

Reviewer: ```suggestion
swift run -c release pix2pix
```

```

> You will need a tensorboard installation in order to run the project. 

Reviewer: With the removal of TensorBoardX, this sentence can probably be removed for now. Again, we'll revisit better TensorBoard support at some point in the future.

## Notes

> - Trainer will emit a lot of intermediate results in a calling directory, you can easily comment out that part. It is a workaround because for some reason tensorboard crashes when provided with result images, I'm investigating on the problem.

Reviewer: Does this note still apply, without the TensorBoard support?

> - You need to specify different URL for `dataset` or we download facades dataset for you
> - This implementation only provides a data loader for facades dataset, but the model is capable of learning on any pix2pix compatible dataset.

Reviewer: ```suggestion
- You need to specify a different URL for `dataset` or this will download the facades dataset for you.
- This implementation only provides a data loader for the facades dataset, but the model is capable of learning on any pix2pix compatible dataset.
```
